### DISCUSSION TITLE
Cue integration with C++

### DISCUSSION THREAD

**Discussion by paragbansal** [cue-lang/cue#2760] [Q&A]
I am trying to explore CUE as configuration management of C++ applications. Objectives are to provide config validation and templating mechanism for users. Easiest way to go about this could be to export CUE configs as JSON and read JSON files in C++ application, but his approach have a problem of CUE schema and application code-base getting out of sync. Therefore, I was exploring the direction of code generating C++ interfaces using the CUE schema and using those interfaces in C++ application, this would ensure that using not-defined or wrongly-defined configuration in schema would lead to compilation failure. To achieve this I would require a deserialisation logic as well to parse the CUE configs and populate those C++ config objects. 

Wanted to get a sense if we have any examples around using CUE in C++ projects or have we explored something like this, or any other approach you would like to suggest! Thank you.

**Comment by kkukshtel** [comment]
As a similar aside, providing a C ABI for core cue functionality would be great for operating/evaluating Cue at application runtime in non-Go applications.
